client/src/components/Settings/SettingsCanvases.res

```diff
@@ -24,10 +29,15 @@ let setInfo = (
   orgs: list<string>,
   orgCanvasList: list<string>,
 ): t => {
+  ..._state,
   canvasList: canvasList,
   username: username,
   orgs: orgs,
   orgCanvasList: orgCanvasList,
 }
 
-let update = (state: t, ()) => state
+let update = (state: t, msg: msg):(t) =>
+  switch msg {
+    | Update(value) => ({...state, newCanvasName: {value: value, error: None}, canvasName:value})
```

Comment:
When you update, can you check if the canvasName is valid (just check for the allowed characters), and if not, show an error. If the canvasName is empty or has an error, can you disable the button?

Reply:
I've tried to check if the canvasName is empty in the validateCanvasName function, but it resulted in a weird behavior where the button gets disabled when the length is 1 rather than 0, so I had to test it inside settingsCanvasesView.

![dark-Dark-Google-Chrome-2022-11](https://user-images.githubusercontent.com/87861924/202779696-b054a221-364b-4e80-adc8-fef9a21539c9.gif)